code/lib/cli/src/autoblock/index.ts

```diff
@@ -0,0 +1,82 @@
+import type { AutoblockOptions, Blocker } from './types';
+import { logger } from '@storybook/node-logger';
+import chalk from 'chalk';
+import boxen from 'boxen';
+import { writeFile } from 'node:fs/promises';
+
+const excludesFalse = <T>(x: T | false): x is T => x !== false;
+
+const blockers: () => BlockerModule<any>[] = () => [
+  // add/remove blockers here
+  import('./block-storystorev6'),
+];
+
+type BlockerModule<T> = Promise<{ blocker: Blocker<T> }>;
+
+export const autoblock = async (
+  options: AutoblockOptions,
+  list: BlockerModule<any>[] = blockers()
+) => {
+  if (list.length === 0) {
+    return null;
+  }
+
+  logger.info('Checking for upgrade blockers...');
+
+  const out = await Promise.all(
+    list.map(async (i) => {
+      const { blocker } = await i;
+      const result = await blocker.check(options);
+      if (result) {
+        return {
+          id: blocker.id,
+          value: true,
+          message: blocker.message(options, result),
+          log: blocker.log(options, result),
+        };
+      } else {
+        return false;
+      }
+    })
+  );
+
+  const faults = out.filter(excludesFalse);
+
+  if (faults.length > 0) {
+    const LOG_FILE_NAME = 'migration-storybook.log';
```

Comment:
Should we add this to the user's gitignore file? That's at least what is discussed here: https://www.notion.so/chromatic-ui/SB8-Release-71d889cb04d240a59e445013d1ecf952?pvs=4#9420b0efd70144e58d5e2e51c6be8305 for `*.storybook.log` files.

Reply:
I'm not really introducing a new file in this PR, it would already have been generated by a failing automigrations precedure before.

Sure we could try to add it to the gitignore file...

But that seems like:
- a task for a separate PR
- more complex than you'd like to believe

Is the user using git?
If the user has no gitignore file yet, do we create it?
Do we just add it to the bottom of the file?

Reply:
Ok. Let's handle this in a separate PR